### ISSUE TITLE
"EEXIST: file already exists..." after Ember-CLI upgrade

### ISSUE THREAD

**Issue by Subaku** [ef4/ember-auto-import#159]
Hello,

I need some guidance on how to debug my build after attempting to upgrade my app to Ember-CLI 3.5.1 from 3.1.4. I keep getting the following during the clean up phase of the build:

```
Build Error (Append)

EEXIST: file already exists, mkdir '/tmp/broccoli-20559DgOk0UjwcA0O/out-403-append_ember_auto_import_analyzer/'


Stack Trace and Error Report: /tmp/error.dump.a138d2bfc58c7703601f799debaac97d.log
```

Looking into that error report we have:

```
ENV Summary:

  TIME: Fri Nov 30 2018 09:41:31 GMT-0800 (PST)
  TITLE: ember
  ARGV:
  - /usr/bin/node
  - /usr/bin/ember
  - build
  EXEC_PATH: /usr/bin/node
  TMPDIR: /tmp
  SHELL: /bin/bash
  PATH:
  - /opt/recon/tv/webapp/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/local/games
  - /usr/games
  PLATFORM: linux x64
  FREEMEM: 702615552
  TOTALMEM: 4158423040
  UPTIME: 106076
  LOADAVG: 0.60205078125,0.36376953125,0.27099609375
  CPUS:
  - Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz - 2494
  - Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz - 2494
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.10.1-DEV
  - http_parser: 2.7.0
  - icu: 58.2
  - modules: 48
  - node: 6.13.0
  - openssl: 1.0.2n
  - uv: 1.16.1
  - v8: 5.1.281.111
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: Error: EEXIST: file already exists, mkdir '/tmp/broccoli-20559DgOk0UjwcA0O/out-403-append_ember_auto_import_analyzer/'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:932:18)
    at Append.build (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/broccoli-append.js:64:32)
    at resolve (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli/lib/wrappers/transform-node.js:29:35)
    at TransformNodeWrapper.build (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli/lib/wrappers/transform-node.js:21:12)
    at Promise.resolve.then.then.then (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli/lib/builder.js:111:28)
    at process._tickCallback (internal/process/next_tick.js:109:7)
  - codeFrame: EEXIST: file already exists, mkdir '/tmp/broccoli-20559DgOk0UjwcA0O/out-403-append_ember_auto_import_analyzer/'
  - errorMessage: EEXIST: file already exists, mkdir '/tmp/broccoli-20559DgOk0UjwcA0O/out-403-append_ember_auto_import_analyzer/'
        at Append (ember-auto-import-analyzer)
-~- created here: -~-
    at Append.Plugin (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli-plugin/index.js:7:31)
    at Append (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/broccoli-append.js:16:9)
    at AutoImport.addTo (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/auto-import.js:71:16)
    at EmberApp.host.addonPostprocessTree (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/auto-import.js:92:29)
    at EmberApp.toTree (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/broccoli/ember-app.js:1772:17)
    at module.exports (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/ember-cli-build.js:91:27)
    at Builder.readBuildFile (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/models/builder.js:52:14)
    at Builder.setupBroccoliBuilder (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/models/builder.js:66:22)
    at new Builder (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/models/builder.js:32:10)
    at BuildTask.run (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/tasks/build.js:15:19)
-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: EEXIST: file already exists, mkdir '/tmp/broccoli-20559DgOk0UjwcA0O/out-403-append_ember_auto_import_analyzer/'
        at Append (ember-auto-import-analyzer)
-~- created here: -~-
    at Append.Plugin (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli-plugin/index.js:7:31)
    at Append (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/broccoli-append.js:16:9)
    at AutoImport.addTo (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/auto-import.js:71:16)
    at EmberApp.host.addonPostprocessTree (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/auto-import.js:92:29)
    at EmberApp.toTree (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/broccoli/ember-app.js:1772:17)
    at module.exports (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/ember-cli-build.js:91:27)
    at Builder.readBuildFile (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/models/builder.js:52:14)
    at Builder.setupBroccoliBuilder (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/models/builder.js:66:22)
    at new Builder (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/models/builder.js:32:10)
    at BuildTask.run (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-cli/lib/tasks/build.js:15:19)
-~- (end) -~-
  - name: BuildError
  - nodeAnnotation: ember-auto-import-analyzer
  - nodeName: Append
  - originalErrorMessage: EEXIST: file already exists, mkdir '/tmp/broccoli-20559DgOk0UjwcA0O/out-403-append_ember_auto_import_analyzer/'
  - stack: Error: EEXIST: file already exists, mkdir '/tmp/broccoli-20559DgOk0UjwcA0O/out-403-append_ember_auto_import_analyzer/'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:932:18)
    at Append.build (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/ember-auto-import/js/broccoli-append.js:64:32)
    at resolve (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli/lib/wrappers/transform-node.js:29:35)
    at TransformNodeWrapper.build (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli/lib/wrappers/transform-node.js:21:12)
    at Promise.resolve.then.then.then (/opt/recon/tv/webapp/src/tvwebapp/ember/reporting-dashboard/node_modules/broccoli/lib/builder.js:111:28)
    at process._tickCallback (internal/process/next_tick.js:109:7)
```

Looking through this myself I can't identify anything obvious. I've also tried running with debugging for ember-auto-import turned on and again nothing weird jumped out at me. Any advice on how to debug this would be appreciated!

Thanks

**Comment by ef4** [comment]
Thanks for reporting. Do you have any customized outputPaths in your ember-cli-build.js?

**Comment by Subaku** [comment]
Thank you for the prompt response!

I do have this:

```
outputPaths: {
    app: {
        js: 'app.js',
        css: {
          app: 'app.css'
        }
    },
    vendor: {
        css: 'vendor.css',
        js: 'vendor.js'
    },
 },
```

**Comment by Subaku** [comment]
Now that you've pointed that out playing around a bit yields the following results.

1) If I remove the entire outputPaths config in my ember-cli-build.js the build doesn't error.
2) If I override an output path with at least one directory level it works as well. Ex: `/foo/vendor.js` instead of just `vendor.js`.

I suppose the question now is how do I allow my builds to not require an extra directory level.

**Comment by ef4** [comment]
Thanks, your bug report made it straightforward for me to find the problem. This should be fixed in 1.2.18.

**Comment by Subaku** [comment]
This fixes it. Thanks!
